src/GafferScene/RendererAlgo.cpp

```diff
@@ -1260,6 +1260,26 @@ struct ObjectOutput : public LocationOutput
 // Public methods for outputting globals.
 //////////////////////////////////////////////////////////////////////////
 
+namespace
+{
+
+ConstOutputPtr addGafferOutputHeaders( const Output *output, const ScenePlug *scene )
+{
+	CompoundDataPtr param = output->parametersData()->copy();
+
+	const Node *node = scene->node();
+	const ScriptNode *script = node->scriptNode();
+
+	// Include the path to the render node to allow tools to back-track from the image
+	param->writable()["header:gaffer:renderNode"] = new StringData(
```

Comment:
I think we definitely need to keep the render node too, as it allows us to more easily restart/repeat renders etc.. from the catalog in the future. If we go to far up, we could end up with ambiguity going back down again.

Reply:
https://github.com/GafferHQ/gaffer/pull/3582/commits/bf02e791c13ad3957089ffb7260f3276a7a0f16d

Reply:
Tom and I discussed this offline and agreed to store the plug rather than the node, but not to backtrack to try to find the output plug that was connected to the render node. This was our thinking :

- Backtracking is still vulnerable to inputs being changed over time. It wouldn't be uncommon to have a node just before the renderer to make IPR-only tweaks, and to be switching the input to that to decide what to render.
- The heuristics involved in trying to backtrack reliably would be fiddly and fragile.
- We do need access to the render node, to pause/restart, as Tom said above.

We've called it `sourceScene` rather than `renderNode` both to reflect the fact that it is a plug, and also that we might want to associate other kinds of source scenes with images, which may not necessarily be render nodes.